CMakeLists.txt

```diff
@@ -78,6 +78,9 @@ if(POLICY CMP0042)
   cmake_policy(SET CMP0042 NEW)
 endif()
 
+set (CMAKE_C_FLAGS "-Wall -Wextra ${CMAKE_C_FLAGS}")
+set (CMAKE_CXX_FLAGS "-Wall -Wextra ${CMAKE_CXX_FLAGS}")
```

Comment:
-1
These should be set outside of cmake, you can set these in the travis script for example, even with Werror if you want.
It's sensitive to the plaftorm/compiler. Even gcc won't yield the same warnings dependings on the version.

Reply:
If I change it only in travis, this will not build with these flags on the local machine using the suggested build method:
* mkdir build; cd build
* cmake ..
* make
Is there a better way?

Reply:
These should not be fixed into the CMakeLists, I think it's better to let each one choose his compiler flags.

Reply:
Ok, I'll put it in the travis file:

`cmake -DCMAKE_INSTALL_PREFIX=~/.local -DNLOPT_MATLAB=OFF -DCMAKE_C_FLAGS='-Wall -Wextra -Werror' -DCMAKE_CXX_FLAGS='-Wall -Wextra -Werror' ..`

.. but only in the first line (not cluttering the file too much) as the warning should fail there already and the mingw builds should not do anything differently, I suppose.

I'll add a commit to the current branch with the fixes. If you like to have a cleaner history I'll squash the commits in a new branch and you may merge from there?